core/node/commitment_generator/src/utils.rs

```diff
@@ -234,3 +241,79 @@ pub(crate) fn convert_vm_events_to_log_queries(events: &[VmEvent]) -> Vec<LogQue
         })
         .collect()
 }
+
+pub(crate) fn pubdata_to_blob_linear_hashes(
+    blobs_required: usize,
+    mut pubdata_input: Vec<u8>,
+) -> Vec<H256> {
+    // Now, we need to calculate the linear hashes of the blobs.
+    // Firstly, let's pad the pubdata to the size of the blob.
+    if pubdata_input.len() % ZK_SYNC_BYTES_PER_BLOB != 0 {
+        let padding =
+            vec![0u8; ZK_SYNC_BYTES_PER_BLOB - pubdata_input.len() % ZK_SYNC_BYTES_PER_BLOB];
+        pubdata_input.extend(padding);
+    }
+
+    let mut result = vec![H256::zero(); blobs_required];
+
+    pubdata_input
+        .chunks(ZK_SYNC_BYTES_PER_BLOB)
+        .enumerate()
+        .for_each(|(i, chunk)| {
+            result[i] = H256(keccak256(chunk));
+        });
+
+    result
+}
+
+pub(crate) async fn read_aggregation_root(
+    connection: &mut Connection<'_, Core>,
+    l1_batch_number: L1BatchNumber,
+) -> anyhow::Result<H256> {
+    // Position of `FullTree::_height` in `MessageRoot`'s storage layout.
+    const AGG_TREE_HEIGHT_KEY: usize = 3;
```

Comment:
AFAICT, some other contract layout constants are defined in the `system_constants` crate (e.g., for `SystemContext`). Maybe, it would make sense to define these constants there as well?